cloudapp/src/app/print-slip-report/print-slip-report.service.ts

```diff
@@ -130,65 +79,16 @@ export class PrintSlipReportService {
 
 
   async findRequestedResources(): Promise<RequestedResource[]> {
-    let pages: Page[] = [
-      new Page(this.circDeskCode, this.libraryCode, 0, this.pageSize, this.restService)
-    ]
-
-    try {
-
-      let totalRecordCount = await pages[0].fetchPage()
-      this.progressChange.emit(0)   // Force the progress spinner animation to start at 0
-      if (totalRecordCount > 0) {
-        pages = this.setupPages(pages[0], totalRecordCount)
-        let pagesIterator: Iterator<Page> = pages.values()
-
-        type PendingPromiseValue = PageFetchValue | void
-        let pendingPromises: Promise<PendingPromiseValue>[] = [ ]
-
-        const addToPendingPromises = (additionalPendingPromises: Promise<PendingPromiseValue>[]) => {
-          pendingPromises = pendingPromises.concat(
-            additionalPendingPromises.map(p => {
-              // Remove the promise from pendingPromises when it resolves
-              let p2 = p.then(x => {
-                let i = pendingPromises.indexOf(p2)
-                pendingPromises.splice(i, 1)
-                return x
-              })
-              return p2
-            })
-          )
-        }
-
-        addToPendingPromises([ pagesIterator.next().value.fetch() ])
 
-        while (pendingPromises.length > 0) {
-          let progress = pages.reduce<number>((acc, page) => acc + page.progress, 0) / pages.length
-          this.progressChange.emit(progress)
-
-          let ret = await Promise.race(pendingPromises)
-          if (ret && 'additionalPendingPromises' in ret) {
-            addToPendingPromises(ret.additionalPendingPromises)
-            let n = pagesIterator.next()
-            if (n.value) {
-              addToPendingPromises([ n.value.fetch() ])
-            }
-          }
-        }
-
-      }
-
-    } catch (err) {
-      this.error.emit(new PrintSlipReportErrorEvent(err))
-      throw err
-    }
-
-    let requestedResources = pages.reduce<RequestedResource[]>(
-      (acc, v) => acc.concat(v.requests),
-      []
+    return this.requestedResourcesService.findRequestedResources(
+      this.circDeskCode,
+      this.libraryCode,
+      this.pageSize,
+      this.progressChange,
+      (count: number) => { this.complete.emit(new PrintSlipReportCompleteEvent(count)) },
+      (err: Error) => { this.error.emit(new PrintSlipReportErrorEvent(err)) },
```

Comment:
Need to be PrintSlipReportError, not Error.

Reply:
Done in commit 986ce86